cty/function/stdlib/datetime.go

```diff
@@ -203,6 +203,33 @@ var FormatDateFunc = function.New(&function.Spec{
 	},
 })
 
+// TimeAddFunc is a function that adds a duration to a timestamp, returning a new timestamp.
+var TimeAddFunc = function.New(&function.Spec{
+	Params: []function.Parameter{
+		{
+			Name: "timestamp",
+			Type: cty.String,
+		},
+		{
+			Name: "duration",
+			Type: cty.String,
+		},
+	},
+	Type: function.StaticReturnType(cty.String),
+	Impl: func(args []cty.Value, retType cty.Type) (cty.Value, error) {
+		ts, err := time.Parse(time.RFC3339, args[0].AsString())
```

Comment:
This is specifically using RFC3339; do you want me to make a `MakeTimeAddFunc(layout string)` to allow/force user to pick a layout ?

Reply:
The `formatdate` function that's already in here has created the precedent that RFC3339 is the canonical time format for `cty`, so I think it makes sense to continue that precedent here. Users that need a different date format as _output_ can pass the result to `formatdate` to change it.

This convention that timestamps are just strings written in RFC3339 notation actually originated in Terraform itself prior to `cty` existing, but I just embraced it here since it seemed like a good enough idea. I wanted to point that out because that is a _Terraform-wide_ convention: provider plugins are expected to _produce_ and _accept_ timestamps in RFC3339 format, even if that means that the provider has to do conversions itself from whatever representation a remote API is using.

Hopefully Packer would be able to dictate a similar dictum so that everything will compose together well, but I don't know if there's existing precedent in Packer for timestamps in other formats. If so, Packer might end up wanting its own time-manipulation functions in order to work well with the existing conventions. I'd prefer to keep the `cty` convention simple either way, and let individual applications handle their own differing conventions if needed.

Reply:
Looking at `formatdate` again reminded me that this package already has a `parseTimestamp` function which wraps `time.Parse(time.RFC3339, ...)` to produce better error messages. It'd be nice to use that here; I think it should be a drop-in replacement.

Reply:
Got it, and I agree. I think I want Packer to do the same thing to get a similar UX.

I just updated formatdate !